src/algebraic_geometry/open_immersion.lean

```diff
@@ -1062,4 +1062,156 @@ end pullback
 
 end LocallyRingedSpace.is_open_immersion
 
+lemma is_open_immersion.open_range {X Y : Scheme} (f : X ⟶ Y) [H : is_open_immersion f] :
+  is_open (set.range f.1.base) := H.base_open.open_range
+
+section open_cover
+
+namespace Scheme
+
+/-- An open cover of `X` consists of a family of open immersions into `X`,
+and for each `x : X` an open immersion (indexed by `f x`) that covers `x`.
+
+This is merely a coverage in the Zariski pretopology, and it would be optimal
+if we could reuse the existing API about pretopologies, However, the definitions of sieves and
+grothendieck topologies uses `Prop`s, so that the actual open sets and immersions are hard to
+obtain. Also, since such a coverage in the pretopology usually contains a proper class of
+immersions, it is quite hard to glue them, reason about finite covers, etc.
+-/
+-- TODO: provide API to and from a preseive.
+structure open_cover (X : Scheme.{u}) :=
+(J : Type u)
+(obj : Π (j : J), Scheme)
+(map : Π (j : J), obj j ⟶ X)
+(f : X.carrier → J)
+(covers : ∀ x, x ∈ set.range ((map (f x)).1.base))
+(is_open : ∀ x, is_open_immersion (map x) . tactic.apply_instance)
+
+attribute [instance] open_cover.is_open
+
+variables {X Y Z : Scheme.{u}} (𝒰 : open_cover X) (f : X ⟶ Z) (g : Y ⟶ Z)
+variables [∀ x, has_pullback (𝒰.map x ≫ f) g]
+
+/-- The affine cover of a scheme. -/
+def affine_cover (X : Scheme) : open_cover X :=
+{ J := X.carrier,
+  obj := λ x, Spec.obj $ opposite.op (X.local_affine x).some_spec.some,
+  map := λ x, ((X.local_affine x).some_spec.some_spec.some.inv ≫
+    X.to_LocallyRingedSpace.of_restrict _ : _),
+  f := λ x, x,
+  is_open := λ x, begin
+    apply_with PresheafedSpace.is_open_immersion.comp { instances := ff },
+    apply_instance,
+    apply PresheafedSpace.is_open_immersion.of_restrict,
+  end,
+  covers :=
+  begin
+    intro x,
+    erw coe_comp,
+    rw [set.range_comp, set.range_iff_surjective.mpr, set.image_univ],
+    erw subtype.range_coe_subtype,
+    exact (X.local_affine x).some.2,
+    rw ← Top.epi_iff_surjective,
+    change epi ((SheafedSpace.forget _).map (LocallyRingedSpace.forget_to_SheafedSpace.map _)),
+    apply_instance
+  end }
+
+instance : inhabited X.open_cover := ⟨X.affine_cover⟩
+
+instance localization_unit_is_iso (R : CommRing) :
+  is_iso (CommRing.of_hom $ algebra_map R (localization.away (1 : R))) :=
+is_iso.of_iso (is_localization.at_one R (localization.away (1 : R))).to_ring_equiv.to_CommRing_iso
+
+instance localization_unit_is_iso' (R : CommRing) :
+  @is_iso CommRing _ R _ (CommRing.of_hom $ algebra_map R (localization.away (1 : R))) :=
+by { cases R, exact algebraic_geometry.Scheme.localization_unit_is_iso _ }
```

Comment:
Not sure where these should go. Probably not here though.

Reply:
Maybe just in the file that defines `CommRing`? Or if localizations aren't available there, in a new file next to it?

Reply:
I think there could be a file in `algebra/category/CommRing/` where we could restate results in `ring_theory` etc in a categorical language.
Do you have a good name for this file? The best I could come up with is `instances.lean`, but there will potentially be things other than instances in the future.

Reply:
hmm, I don't have a better name for now. But changing a filename is cheap, so I would go with your proposal for now.